src/client/Microsoft.Identity.Client/OAuth2/OAuth2Client.cs

```diff
@@ -73,10 +78,14 @@ public async Task<MsalTokenResponse> GetTokenAsync(Uri endPoint, RequestContext
             return await ExecuteRequestAsync<MsalTokenResponse>(endPoint, HttpMethod.Post, requestContext).ConfigureAwait(false);
         }
 
-        internal async Task<T> ExecuteRequestAsync<T>(Uri endPoint, HttpMethod method, RequestContext requestContext, bool expectErrorsOn200OK = false)
+        internal async Task<T> ExecuteRequestAsync<T>(Uri endPoint, HttpMethod method, RequestContext requestContext, bool expectErrorsOn200OK = false, bool sendDeviceAuthParams = true)
```

Comment:
As discussed offline, this is not the right place to invoke the pkauth manager because of 2 reasons: 

- this class is responsible for making all calls to STS - /token, instance, oidc, ws-trust, /device_code
- Single Responsibility Principle - this low level class should not gain another responsibility, it's already doing a lot...

So my proposal is to move this logic to TokenClient.cs which handles communication with /token endpoint only.

Reply:
Lets discuss this one offline, i ran into some uncertainty while trying to make this change